internationalisation (2)

pbranly opened the pull request.

pbranly committed:
internationalisation

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
Internationalisations

pbranly committed:
internationalisation

pbranly committed:
internationalisations

pbranly committed:
internationalisations

pbranly committed:
Internationalisations

pbranly committed:
internationalisations

pbranly committed:
Add TranslatedStrings_en.txt for iNaturalist plugin

Added English translation strings for iNaturalist plugin.

pbranly committed:
Add French translations for iNaturalist plugin

Added French translations for various iNaturalist plugin strings.

pbranly reviewed (commented):
Hi seems ok now

pbranly commented:
Hey 
Tests seem positive this time !
I spent more time to verify and verify 
Cheers 
Philippe 
From France during a big storm

pbranly commented:
Hi 
Will you merge it ?
Phil

pbranly commented:
Hi 
Will you ?
Phil

rcloran commented on lr-inaturalist-publish.lrdevplugin/ExportServiceProvider.lua:
```diff
@@ -265,11 +273,20 @@ function exportServiceProvider.sectionsForTopOfDialog(f, propertyTable)
 						if propertyTable.syncKeywordsCommon then
-							r = "Set scientific name as a keyword synonym"
+							r = LOC(
+								"$$/iNat/Export/SyncSynonymScientific=Set scientific name as a keyword " .. "synonym"
```
Remove unnecessary string concatenation here

pbranly replied:
Hi
Yes there is an unecessary test in my code that I have to correct; 
nevertheless if i concatenate this line, I am afraid it will overflow the maximum length and surely generate a stylua or/and a luacheck error. What do you think ?

pbranly replied:
modified

rcloran commented on lr-inaturalist-publish.lrdevplugin/ExportServiceProvider.lua:
```diff
@@ -403,52 +425,87 @@ local function checkSettings(settings)
-			table.insert(suggestions, " - Reduce long edge size to 2048 pixels or fewer")
+			table.insert(
+				suggestions,
+				LOC("$$$/iNat/Export/SuggestionReduceLongEdge= - Reduce long edge size to 2048 pixels or " .. "fewer")
```
Remove unnecessary string concatenation.

pbranly replied:
if i concatenate this line, I am afraid it will overflow the maximum length and surely generate a stylua or/and a luacheck error. What do you think ?

pbranly replied:
modified

rcloran commented on lr-inaturalist-publish.lrdevplugin/ExportServiceProvider.lua:
```diff
@@ -403,52 +425,87 @@ local function checkSettings(settings)
+			table.insert(
+				suggestions,
+				LOC(
+					'$$$/iNat/Export/SuggestionLimitMP= - If you need to use "Megapixels", limit to less ' .. "than 4.2"
```
Remove unnecessary string concatenation

pbranly replied:
if i concatenate this line, I am afraid it will overflow the maximum length and surely generate a stylua or/and a luacheck error. What do you think ?

pbranly replied:
done

rcloran commented on lr-inaturalist-publish.lrdevplugin/ExportServiceProvider.lua:
```diff
@@ -403,52 +425,87 @@ local function checkSettings(settings)
-		table.insert(suggestions, ' - Use "Long Edge" instead of "Percentage"')
+		table.insert(
+			suggestions,
+			LOC('$$$/iNat/Export/SuggestionLongEdgeNotPercent= - Use "Long Edge" instead of ' .. '"Percentage"')
```
Remove unnecessary string concatenation.

pbranly replied:
if i concatenate this line, I am afraid it will overflow the maximum length and surely generate a stylua or/and a luacheck error. What do you think ?

pbranly replied:
modified

pbranly commented:
Hi
 thank you for the deep analysis; Please review my comments.
 Will it be the end of the tunnel ? 
 Phil

rcloran commented:
In all the places I have commented, you have written (on one line) `"foo" .. "bar"` instead of simply writing `"foobar"`. Changing this will make it easier to read and make the line *shorter*, not longer, so your concern about line length is moot.

pbranly committed:
concatenation corrections

pbranly committed:
concatenation issue

pbranly committed:
Refactor localization function calls for keywordsconcatenation

pbranly commented:
OK well seen !
I made the changes on my side; I hope you got them
Phil

rcloran commented on lr-inaturalist-publish.lrdevplugin/ExportServiceProvider.lua:
```diff
@@ -276,10 +276,8 @@ function exportServiceProvider.sectionsForTopOfDialog(f, propertyTable)
+								"$$/iNat/Export/SyncSynonymScientific=Set scientific name as a keyword synonym"
 							)
-						else
-							r = LOC("$$/iNat/Export/SyncSynonymCommon=Set common name as a keyword synonym")
```
Huh, why did you delete this?

pbranly replied:
Hi because it was not in your code and hided a variable declaration forgotten

rcloran commented on lr-inaturalist-publish.lrdevplugin/TranslatedStrings_en.txt:
```diff
@@ -0,0 +1,107 @@
+"$$$/iNat/Export/AccountTitle=iNaturalist Account"
+"$$$/iNat/Export/LogInButton=Log in"
+"$$$/iNat/Export/OptionsTitle=Export options"
+"$$$/iNat/Export/SpeciesGuessInfo=Set species guess from keywords within"
```
I've already provided review feedback (on an older PR) that this translation is only part of the string, and will break the UI.

pbranly replied:
seems ok now nevertheless I have now a stylua I cant explain; what a nightmare ... thanks for your patience

pbranly replied:
Right sorry about that; there are still several lines with errors that I have to correct

rcloran reviewed (changes requested):
I see that we still have the problem with incomplete translations because of the buggy AI script that you've used. I only saw one of them and stopped reading your changes at that point -- I'm guessing all the others are still there. And because you insist on closing PRs and opening a new one so often, we've lost a bunch of that history.

I'm not interested in doing the same reviews over and over again.

pbranly committed:
variable correction

pbranly committed:
multiline corrections

pbranly committed:
multiline corrections

pbranly committed:
Refine translated strings in TranslatedStrings_en.txt

pbranly committed:
Refine translation strings in TranslatedStrings_en.txt

Updated translation strings for clarity and added details.

pbranly committed:
Update French translations in TranslatedStrings_fr.txt

Updated French translations for various strings in the iNaturalist plugin.

pbranly committed:
Refactor localization strings in ExportServiceProvider

pbranly commented:
Hi
Everything seems (!) correct now. Or at least on my side 
The last chance ….
Thanks
Phil

rcloran reviewed (changes requested):
Some translation strings are still incomplete.

pbranly commented:
> Some translation strings are still incomplete.

Hi
This is a nightmare and again thank you for your patience 
Do you have an exemple ? Because I made so many verifications that I can't see the end of the tunnel now 
Thanks 
Phil

rcloran commented:
> > Some translation strings are still incomplete.
> 
> Hi This is a nightmare

It seems you're trying to rely on scripts and AI to do the work. It would be much easier if you just did the work yourself.

> Do you have an exemple ?

Yes, I already commented on it. In the English translation file you have `iNat/Export/SpeciesGuessInfo` as "Set species guess from keywords within", but in the source code it is 'Set species guess from keywords within "^1" keyword\n(The setting for which keyword is in the "Synchronization" section)'.

You fixed it in 127c55df6a3cb983f01d810855ce44352a80e608, and then broke it again in 4d532bccbcf0ad7c5c65cde7fa6b36e53caf8cd9.

pbranly commented:
> > > Some translation strings are still incomplete.
> > 
> > 
> > Hi This is a nightmare
> 
> It seems you're trying to rely on scripts and AI to do the work. It would be much easier if you just did the work yourself.
> 
> > Do you have an exemple ?
> 
> Yes, I already commented on it. In the English translation file you have `iNat/Export/SpeciesGuessInfo` as "Set species guess from keywords within", but in the source code it is 'Set species guess from keywords within "^1" keyword\n(The setting for which keyword is in the "Synchronization" section)'.
> 
> You fixed it in [127c55d](https://github.com/rcloran/lr-inaturalist-publish/commit/127c55df6a3cb983f01d810855ce44352a80e608), and then broke it again in [4d532bc](https://github.com/rcloran/lr-inaturalist-publish/commit/4d532bccbcf0ad7c5c65cde7fa6b36e53caf8cd9).

you are right my tools correct but generate errors
by hand, I am sure I will generate more mistakes !
again thanks for your patience 
I will verify again before (rererere)submitting
Phil

pbranly committed:
Refine translated strings in TranslatedStrings_en.txt

Updated various strings for clarity and detail in the iNaturalist Publish Plugin.

pbranly committed:
Update French translations in TranslatedStrings_fr.txt

pbranly commented:
Hi
Any news ?
Thanks
Phil

pbranly commented:
Hi
I think you will not activate that PR.
Nevertheless I wish you a merry Christmas and a happy new year 
Phil
